src/main/java/com/jfrog/ide/idea/scan/ScanManagersFactory.java

```diff
@@ -246,26 +246,29 @@ private void createScanManagersForPackageDirs(Set<String> packageDirs, Map<Integ
      * We create a new instance to allow setting the scan results separately after the Xray scan.
      * On the other hand, the scan cache map is a single map shared between all scanners.
      *
-     * @param scanManagers - The scan managers before Xray scan
+     * @param scanManagers  - The scan managers before Xray scan
+     * @param scanLogicType - /graph/scan or /summary/component logic
+     * @param quickScan     - True to allow usage of the scan cache.
      * @throws IOException in case of any I/O error.
      */
-    private void setScanLogic(Map<Integer, ScanManager> scanManagers, Utils.ScanLogicType scanLogicType) throws IOException {
-        ScanCache scanCache = createXrayScanCache();
+    private void setScanLogic(Map<Integer, ScanManager> scanManagers, Utils.ScanLogicType scanLogicType, boolean quickScan) throws IOException {
+        ScanCache scanCache = createXrayScanCache(quickScan);
         Logger logger = Logger.getInstance();
         scanManagers.values().forEach(manager -> manager.setScanLogic(createScanLogic(scanLogicType, scanCache, logger)));
     }
 
     /**
      * Create the scan cache object and the directories needed for it.
      *
+     * @param quickScan - True to load the scan cache.
      * @return scan cache.
      * @throws IOException in cace of any I/O error.
      */
-    private ScanCache createXrayScanCache() throws IOException {
+    private ScanCache createXrayScanCache(boolean quickScan) throws IOException {
```

Comment:
I think that here the `quickScan` is a bit confusing. What do you think about something like `loadFromDisk` or `initNew`?